### ISSUE TITLE
API wheel client throws exception and success=true

### ISSUE THREAD

**Issue by amendlik** [saltstack/salt#38540]
### Description of Issue/Question
API exceptions do not affect anything in the response other that the return string, making it difficult to catch a failed call.

### Steps to Reproduce Issue
Here is an example of a bad request. It is deliberately missing a required keyword argument.
Request:
```
[
  {
    "client": "wheel",
    "fun": "key.finger"
  }
]
```
Response:
```
{
  "return": [
    {
      "tag": "salt/wheel/20161229061105297133",
      "data": {
        "jid": "20161229061105297133",
        "return": "Exception occurred in wheel key.finger: Traceback (most recent call last):\n  File \"/home/amendlik/source/salt/salt/client/mixins.py\", line 364, in _low\n    expected_extra_kws=CLIENT_INTERNAL_KEYWORDS\n  File \"/home/amendlik/source/salt/salt/utils/__init__.py\", line 1054, in format_call\n    used_args_count\nSaltInvocationError: finger takes at least 1 argument (0 given)\n",
        "success": true,
        "_stamp": "2016-12-29T13:11:05.901728",
        "tag": "salt/wheel/20161229061105297133",
        "user": "admin",
        "fun": "wheel.key.finger"
      }
    }
  ]
}
```
Notice it still returns a 200, and (worse) it contains “success”: true. Short of reading the “return” string, there is no way for a caller to tell that this function failed. 
### Versions Report
Salt Version:
           Salt: 2016.11.1
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: 3.2.2
       dateutil: 1.4.1
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: 0.20.0
        libnacl: Not Installed
       M2Crypto: 0.20.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: 0.20.3
         Python: 2.6.6 (r266:84292, Aug  9 2016, 06:11:56)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: redhat 6.8 Santiago
        machine: x86_64
        release: 2.6.32-642.11.1.el6.x86_64
         system: Linux
        version: Red Hat Enterprise Linux Server 6.8 Santiago

**Comment by whiteinge** [comment]
RunnerClient currently sets the `success` field to `False` for traceback (I think), and `LocalClient` does for most minion-side tracebacks (though some slip through from time to time, IME). It looks like `WheelClient` needs to set that value when catching tracebacks.

**Comment by whiteinge** [comment]
Thanks, @amendlik.

**Comment by terminalmage** [comment]
@whiteinge Looking at salt/client/mixins.py (location identified in the traceback), it is setting ``data['success']`` to ``False``. So that suggests that salt-api is ignoring the ``success`` key in the return data. I'll continue to dig.

**Comment by terminalmage** [comment]
@whiteinge Think I have a fix for this in https://github.com/saltstack/salt/pull/38925.

**Comment by terminalmage** [comment]
Also added a fix for https://github.com/saltstack/salt/issues/38537 to https://github.com/saltstack/salt/pull/38925

**Comment by whiteinge** [comment]
Good fixes. Thanks for tracking those down!

**Comment by whiteinge** [comment]
@amendlik see the above PR courtesy of some @terminalmage magic. It's now merged so you can test on the latest 2016.3 branch if you'd like, or they'll be in the next point-release.
